ai-catalog/samples/gemini-live-todo/src/main/java/com/android/ai/samples/geminilivetodo/ui/TodoScreen.kt

```diff
@@ -0,0 +1,285 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.android.ai.samples.geminilivetodo.R
+import com.android.ai.samples.geminilivetodo.data.Todo
+import kotlin.collections.reversed
+
+/**
+ * The main screen for the To-do list application.
+ * This composable is stateful, connecting to the ViewModel to manage UI state and events.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TodoScreen(viewModel: TodoScreenViewModel = hiltViewModel()) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    var text by remember { mutableStateOf("") }
+
+    val activity = LocalActivity.current as Activity
+
+    LaunchedEffect(Unit) {
+        viewModel.initializeGeminiLive(activity)
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                colors = TopAppBarDefaults.topAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    titleContentColor = MaterialTheme.colorScheme.onPrimaryContainer,
+                ),
+                title = { Text(stringResource(R.string.gemini_live_title)) },
+            )
+        },
+        floatingActionButton = {
+            MicButton(
+                uiState = uiState,
+                onToggle = { viewModel.toggleLiveSession(activity) },
+            )
+        },
+        floatingActionButtonPosition = FabPosition.Center,
+        modifier = Modifier.fillMaxSize(),
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier
+                .padding(paddingValues)
+                .padding(16.dp)
+                .imePadding()
+                .fillMaxSize(),
+        ) {
+            TodoInput(
+                text = text,
+                onTextChange = { text = it },
+                onAddClick = {
+                    viewModel.addTodo(text)
+                    text = ""
+                },
+            )
+
+            when (uiState) {
+                is TodoScreenUiState.Initial -> {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        contentAlignment = Alignment.Center,
+                    ) {
+                        CircularProgressIndicator()
+                    }
+                }
+                is TodoScreenUiState.Success -> {
+                    val todos = (uiState as TodoScreenUiState.Success).todos
+                    LazyColumn(modifier = Modifier.fillMaxSize()) {
```

Comment:
I just noticed that this section is completely copy pasted, you should extract it. But also, I don't think you even need the higher level UIState error, you never use it and you have the error in the LiveSessionState